app/models/course.rb

```diff
@@ -442,6 +443,12 @@ def ensure_modern_languages
     end
   end
 
+  def ensure_site_statuses_match_study_mode
+    site_statuses.select(&:with_vacancies?).each do |site_status|
```

Comment:
I know this is merged, but just to say there's a scope `with_vacancies` that could've, and I think probably should've, been used here. Just a minor performance issue.